frontend/public/components/cluster-service-classes.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+
+import { ColHead, DetailsPage, List, ListHeader, ListPage, ResourceRow } from './factory';
+import { SectionHeading, detailsPage, navFactory, ResourceLink, ResourceSummary } from './utils';
+// eslint-disable-next-line no-unused-vars
+import { K8sResourceKind, K8sResourceKindReference } from '../module/k8s';
+import * as _ from 'lodash-es';
+
+const ServiceClassesReference: K8sResourceKindReference = 'ClusterServiceClass';
+
+const normalizeIconClass = (iconClass) => {
+  return _.startsWith(iconClass, 'icon-') ? `font-icon ${iconClass}` : iconClass;
+};
+
+export const ServiceClassIcon: React.SFC<ServiceClassIconProps> = ({iconClass, imageUrl, imageClass}) => {
+  if (iconClass) {
+    return <span className={`fa-3x co-cluster-service-class-title__icon ${normalizeIconClass(iconClass)}`} />;
+  }
+  if (imageUrl) {
+    return <img className={`co-cluster-service-class-title__icon ${imageClass}`} src={imageUrl} />;
+  }
+};
+ServiceClassIcon.displayName = 'ServiceClassIcon';
+
+const ServiceClassHeader: React.SFC<ServiceClassHeaderProps> = props => <ListHeader>
+  <ColHead {...props} className="col-lg-5 col-md-5 col-sm-6 col-xs-6" sortField="metadata.name">Name</ColHead>
+  <ColHead {...props} className="col-lg-4 col-md-4 col-sm-6 col-xs-6" sortField="spec.externalMetadata.displayName">Display Name</ColHead>
+  <ColHead {...props} className="col-lg-3 col-md-3 hidden-sm hidden-xs" sortField="spec.externalName">External Name</ColHead>
+</ListHeader>;
+
+const ServiceClassListRow: React.SFC<ServiceClassesRowProps> = ({obj: serviceClass}) => <ResourceRow obj={serviceClass}>
+  <div className="col-lg-5 col-md-5 col-sm-6 col-xs-6 co-resource-link-wrapper">
+    <ResourceLink kind="ClusterServiceClass" name={serviceClass.metadata.name} />
+  </div>
+  <div className="col-lg-4 col-md-4 col-sm-6 col-xs-6">
+    <ServiceClassIcon iconClass={_.get(serviceClass, ['spec', 'externalMetadata', 'console.openshift.io/iconClass'])}
+      imageUrl={_.get(serviceClass, ['spec', 'externalMetadata', 'imageUrl'])}
+      imageClass="co-cluster-service-class-row__img" />
+    {_.get(serviceClass, 'spec.externalMetadata.displayName', '-')}
```

Comment:
There's an edge case here where we might have an icon with no display name, which would look weird

Reply:
Ok, I agree, what would you suggest the right thing to do here?